pages/docs/reference/evolution/migrating-multiplatform-project-to-14.md

```diff
@@ -0,0 +1,187 @@
+
+### Follow the default libraries’ layout 
+
+The layout of Kotlinx has changed and now corresponds to the default layout, which we recommend using:
+The '“root” or “umbrella” library module now has a name without a suffix (e.g. `kotlinx-coroutines-core` instead of `kotlinx-coroutines-core-native`). Publishing libraries with [maven-publish Gradle plugin](https://docs.gradle.org/current/userguide/publishing_maven.html) follows this layout by default.
+
+### Migrate to the hierarchical project structure
+
+A hierarchical project structure allows reusing code in similar targets, as well as publishing and consuming libraries with granular APIs targeting similar platforms. We recommend enabling the hierarchical project structure in your libraries when you migrating to Kotlin 1.4.0:
+
+* Libraries published with the hierarchical project structure enabled are compatible with all kinds of projects, both with and without the hierarchical project structure enabled. However, libraries published without the hierarchical project structure can’t be used in a shared-native source set. So, for example, users with `ios()` shortcuts in their `gradle.build` files won’t be able to use your library in their iOS-shared code.
+* Starting from Kotlin-1.4.20, the hierarchical project structure with the usage of platform-dependent libraries in shared source sets will be enabled by default in multiplatform projects. So the sooner you support it, the sooner users will be able to migrate. We’ll also be very thankful if you report any bugs you find to our [issue tracker.](http://kotl.in/issue) 
+
+To enable hierarchical project structure support, add the following to your `gradle.properties`:
+
+<div class="sample" markdown="1" theme="idea" mode='xml'>
+
+```
+kotlin.mpp.enableGranularSourceSetsMetadata=true
+```
+
+</div>
+
+## For build authors
+
+### Check task names
+
+The introduction of the hierarchical project structure in multiplatform projects resulted in a couple of changes to the names of some Gradle tasks:
+
+* The `metadataJar` task has been renamed to `allMetadataJar`
+* There are new `compile<SourceSet>KotlinMetadata` tasks for all published intermediate source-sets
+
+These changes are relevant only for projects with the hierarchical project structure enabled.
+
+## For using the Kotlin/JS target
+
+### Changes related to npm dependency management
+
+When declaring dependencies on npm packages, you are now required to explicitly specify a version or version range based on [npm’s semver syntax](https://docs.npmjs.com/misc/semver#versions). Specifying multiple version ranges is also supported.
+
+While we don’t recommend it, you can use a wildcard `*` in place of a version number if you do not want to specify a version or version range explicitly.
```

Comment:
`While we don’t recommend it, you can` is not the best wording for docs =) Everything written in docs is recommended by default, unless we're describing a common issue that people face (`Doing so is not recommended`).
If we have a doc page on this topic, maybe add a link? And perhaps an example of new syntax here would be nice.

Reply:
@SebastianAigner PTAL